packages/evm/contracts/treasury/ProtocolTreasury.sol

```diff
@@ -49,26 +50,27 @@ contract ProtocolTreasury is ControlledRecoverable, ITreasury {
     }
 
     /**
-    * @notice Withdraw `@tokenAmount(_token, _amount)` from sender to `_to`
+    * @notice Withdraw `@tokenAmount(_token, _amount)` from `_from` to `_to`
     * @param _token ERC20 token to be withdrawn
+    * @param _from Address withdrawing the tokens from
     * @param _to Address of the recipient that will receive the tokens
     * @param _amount Amount of tokens to be withdrawn from the sender
     */
-    function withdraw(IERC20 _token, address _to, uint256 _amount) external {
-        _withdraw(_token, msg.sender, _to, _amount);
+    function withdraw(IERC20 _token, address _from, address _to, uint256 _amount) external authenticate(_from) {
+        _withdraw(_token, _from, _to, _amount);
     }
 
     /**
-    * @notice Withdraw all the tokens from `_to` to themself
+    * @notice Withdraw all the tokens from `_from` to themself
     * @param _token ERC20 token to be withdrawn
-    * @param _to Address of the recipient that will receive their tokens
+    * @param _from Address withdrawing the tokens from and where the tokens will be transferred
     */
-    function withdrawAll(IERC20 _token, address _to) external {
-        IConfig config = _protocolConfig();
-        require(config.areWithdrawalsAllowedFor(_to), ERROR_WITHDRAWS_DISALLOWED);
+    function withdrawAll(IERC20 _token, address _from) external {
+        bool canWithdrawAll = _isSignatureValid(_from) || _protocolConfig().areWithdrawalsAllowedFor(_from);
```

Comment:
I'm not sure I fully understand the difference between `withdraw()` and `withdrawAll()`; why does this one check `_protocolConfig().areWithdrawalsAllowedFor(_from)`?

Furthermore, I'm wondering why this is an `or` and not an `and`. Wouldn't you want to check that the originating account has allowed this whether it's the direct caller or not?

-------

It's a bit more gas, but I'm also wondering if it'd be worth still using the `authenticate()` modifier here.

In this case, I think the signature being wrong should present a different error.